From 7363d0ec5c84e873d86defc8a02533e6debcb4bd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=A9r=C3=A9my=20Bobbio?= <lunar@debian.org>
Date: Wed, 27 Jan 2016 17:43:44 +0100
Subject: [PATCH 1/2] do not set RPATH for tlsh_unittest

Using RPATH leads to problem when linked libraries have their own
dependencies. See: https://wiki.debian.org/RpathIssue
---
 test/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index bc92ffc..8b9181f 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -26,3 +26,4 @@ add_executable(tlsh_unittest tlsh_unittest.cpp)
 target_link_libraries(tlsh_unittest tlsh_shared)
 set_target_properties(tlsh_unittest PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR}/bin)
 set_target_properties(tlsh_unittest PROPERTIES OUTPUT_NAME tlsh_unittest${BUILD_POSTFIX})
+set_target_properties(tlsh_unittest PROPERTIES SKIP_BUILD_RPATH TRUE)
-- 
2.7.0


From 9f6f94efe8c900a522607e2cbf7f2d443511c3b5 Mon Sep 17 00:00:00 2001
From: anthraxx <levente@leventepolyak.net>
Date: Wed, 27 Jan 2016 19:01:12 +0100
Subject: [PATCH 2/2] set LD_PRELOAD for running tests against local shared
 library

---
 Testing/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Testing/CMakeLists.txt b/Testing/CMakeLists.txt
index 42c739f..c7276b8 100644
--- a/Testing/CMakeLists.txt
+++ b/Testing/CMakeLists.txt
@@ -3,3 +3,4 @@
 # instead
 add_test(tlsh_unittest_len "${CMAKE_SOURCE_DIR}/Testing/test.sh")
 add_test(tlsh_unittest_xlen "${CMAKE_SOURCE_DIR}/Testing/test.sh" "-xlen")
+set_tests_properties(tlsh_unittest_len tlsh_unittest_xlen PROPERTIES ENVIRONMENT "LD_PRELOAD=${CMAKE_SOURCE_DIR}/lib/libtlsh.so.0")
-- 
2.7.0

